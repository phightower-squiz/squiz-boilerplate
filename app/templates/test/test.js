var grunt = require('grunt');
var _     = require('lodash');
var path  = require('path');
var config = grunt.config('config');

function fileNotEmpty(file) {
    var content = grunt.file.read(file);
    return !content.match(/^[\s\n\t]*$/);
}

module.exports = {

    // Make sure we are set up correctly
    copy: function (test) {
        test.ok(grunt.file.exists(config.source + '/html/_test.html'), 'The test file has been created');
        test.done();
    },

    output: function (test) {
        var outputFile = config.dest + '/_test.html';
        var output     = grunt.file.read(outputFile);
        var expected   = grunt.file.read('test/expected/test.html');

        test.ok(fileNotEmpty(outputFile), 'The output has content');
        test.equal(output, expected, 'The output HTML matches the expected output');

        // Test that we have expected file outputs
        _.each(['styles/test/main.css',
                'styles/test/test1.css',
                'styles/test/test2.css',
                'js/test/vendor/jquery.min.js',
                'js/test/vendor/modernizr.min.js',
                'js/test/global.js'],
        function (file) {
            var fullPath = path.join(config.dest, file);
            test.ok(grunt.file.exists(fullPath), 'The file ' + file + ' should exist');
            test.ok(fileNotEmpty(fullPath), 'The file ' + file + ' should not be empty');
        });

        test.done();
    }
}