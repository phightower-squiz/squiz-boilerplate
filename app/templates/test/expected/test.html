<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8" />
        <!--[if IE]><meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" /><![endif]-->
        <title>boilerplate-test - (version 0.0.0)</title>
        <meta name="viewport" content="width=device-width, initial-scale=1.0">

        <link rel="shortcut icon" href="files/favicon.ico">
        <link rel="apple-touch-icon-precomposed" href="files/logo-144x144.png">

        <link rel="stylesheet" href="styles/test/main.css"/>

        <link rel="stylesheet" href="styles/test/test3.css" media="screen" />
        
        <script src="js/test/vendor/modernizr.min.js"></script>
    </head>
    
    <body class="no-js">
    
    <div id="page-wrapper">
    <!--noindex-->
        <div class="imported-content">This content has been imported</div>
        <div class="header" role="banner">

        </div><!-- /.header -->
    <!--endnoindex-->
            <!-- start main-content -->
            <div id="main" role="main">

            </div><!-- /#main -->
            <!-- end main-content -->
    <!--noindex-->
        <div class="footer" role="contentinfo">

        </div><!-- /.footer -->
    <!--endnoindex-->
    </div><!-- /#page-wrapper -->
    <script src="js/test/vendor/jquery.min.js"></script>
    <script src="js/test/global.js"></script>

    </body>
</html>